eclipse-collections-code-generator/src/main/resources/impl/map/immutable/immutablePrimitiveObjectSingletonMap.stg

```diff
@@ -852,7 +852,9 @@ final class Immutable<name>ObjectSingletonMap\<V> extends AbstractImmutable<name
     @Override
     public \<S> ImmutableBag\<Pair\<V, S>\> zip(Iterable\<S> that)
     {
-        return this.zip(that, HashBag.\<Pair\<V, S>\>newBag()).toImmutable();
```

Comment:
This might be ok to not pre-size in this file. Since the max the zipped Bag will be is of size 1. Pre-sizing it to 1 defaults it to 10.

Reply:
As far as I see in objectPrimitiveHashMap.stg pre-sizing to 1 gives map capacity=2 while without presizing it gives capacity=16.

Reply:
@nikhilnanivadekar what kind of instance check do you mean? can you provide an example?